modules/http2/src/main/java/org/glassfish/grizzly/http2/AlpnSupport.java

```diff
@@ -37,43 +36,24 @@
 import org.glassfish.grizzly.npn.AlpnClientNegotiator;
 import org.glassfish.grizzly.npn.AlpnServerNegotiator;
 import org.glassfish.grizzly.npn.NegotiationSupport;
-import org.glassfish.grizzly.ssl.HandshakeListener;
 import org.glassfish.grizzly.ssl.SSLBaseFilter;
+import org.glassfish.grizzly.ssl.SSLBaseFilter.HandshakeListener;
 import org.glassfish.grizzly.ssl.SSLUtils;
 
 /**
  * Grizzly TLS Next Protocol Negotiation support class.
- * 
  */
 public class AlpnSupport {
-    private final static Logger LOGGER = Grizzly.logger(AlpnSupport.class);
+    private static final Logger LOGGER = Grizzly.logger(AlpnSupport.class);
 
-    private final static Map<SSLEngine, Connection<?>> SSL_TO_CONNECTION_MAP =
-            new WeakHashMap<>();
-    
+    private static final Map<SSLEngine, Connection<?>> SSL_TO_CONNECTION_MAP = new WeakHashMap<>();
     private static final AlpnSupport INSTANCE;
-    private static final Method nativeHandshakeMethod;
-    
-    static {
-        boolean isExtensionFound = false;
```

Comment:
When I started to test combinations of JDK vs. grizzly vs. npn-bootstrap vs. npn-api, AlpnSupport sometimes threw exceptions on unexpected places. See Confluence page for few details.
Finally, I created a new class for all this heuristic and tested it against all JDK8 versions I have (both AdoptJDK and Zulu 8u265 are the newest)